src/Microsoft.Health.Fhir.Core/Features/Search/Converters/FhirTypedElementToSearchValueConverterManager.cs

```diff
@@ -36,5 +41,33 @@ public bool TryGetConverter(string fhirType, Type searchValueType, out ITypedEle
 
             return _converterDictionary.TryGetValue((fhirType, searchValueType), out converter);
         }
+
+        internal class ExtensionConverter<T> : ITypedElementToSearchValueConverter
```

Comment:
><T> [](http://example.com/codeflow?start=41&length=3)

Can we add comment why it's generic?

Reason why it's generic is to not add this converter to DI.

[https://github.com/microsoft/healthcare-shared-components/blob/1e0dc8532113a5f7fdd0692c922e91b8273c240e/src/Microsoft.Health.Extensions.DependencyInjection/TypeRegistrationExtensions.cs#L23](https://github.com/microsoft/healthcare-shared-components/blob/1e0dc8532113a5f7fdd0692c922e91b8273c240e/src/Microsoft.Health.Extensions.DependencyInjection/TypeRegistrationExtensions.cs#L23)

we use this method to inject all interface implemenations into DI, and one of the checks to add it - is this class generic or not.

Reply:
This doesn't sound like a good reason to use a Generic... This problem also exists in other places too where there is a class that shouldn't be included in the scanning convention.

The simple solution would just be at add this to a `.Where` clause to exclude it:
https://github.com/microsoft/fhir-server/blob/1c5cd55a505a1f98cd82d3532fa7e198a713af15/src/Microsoft.Health.Fhir.Shared.Api/Modules/SearchModule.cs#L92-L95

A broader solution might be to somehow use an attribute or tag for `[NoScanningRegistration]` (I don't like the name), but thoughts?

Reply:
Made `ExtensionConverter` non-generic and added a `.Where` clause in 51adeb6.

Using an attribute is an interesting idea... I created a story to investigate ([AB#82937](https://microsofthealth.visualstudio.com/Health/_workitems/edit/82937)).